worker/uniter/modes.go

```diff
@@ -207,11 +223,15 @@ func ModeAbide(u *Uniter) (next Mode, err error) {
 // is in an Alive state.
 func modeAbideAliveLoop(u *Uniter) (Mode, error) {
 	for {
+		if err := setAgentStatus(u, params.StatusIdle, "", nil); err != nil {
```

Comment:
I'm worried about this -- ModeAbide doesn't imply idle. A few seconds without some other channel triggerinng does imply idle, but just entering ModeAbide doesn't.